File: _courses/spring-data-jpa-book/src/main/java/io/beaniejoy/springdatajpabook/entity/Member.java
package io.beaniejoy.springdatajpabook.entity;

import javax.persistence.*;

@Entity
@NamedQuery(
        name = "Member.findAllByName",
        query = "select m from Member m where m.name = :name and m.email like '%@example.com'")
@Table(name = "member")
public class Member {

    @Id @GeneratedValue
    @Column(name = "MEMBER_ID")
    private Long id;

    private String name;

    private String email;

    public Member() {
    }

    public Member(Long id, String name, String email) {
        this.id = id;
        this.name = name;
        this.email = email;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }
}

File: _courses/junit-java-test/src/main/java/io/beaniejoy/junittest/domain/Member.kt
package io.beaniejoy.junittest.domain

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
class Member protected constructor(
    id: Long,
    email: String?
) {
    @Id
    @GeneratedValue
    val id: Long = id

    var email: String? = email
        protected set

    companion object {
        // 편의상 id 값도 인자로 포함
        fun createMember(id: Long, email: String): Member {
            return Member(id, email)
        }
    }
}